src/whidbey/plugin.clj

```diff
@@ -1,29 +1,98 @@
 (ns whidbey.plugin
+  "This namespace runs inside of Leiningen and rewrites the project map to
+  include the customizations provided by Whidbey."
   (:require
     [leiningen.core.project :as project]))
 
 
-(defn whidbey-profile
-  [options]
-  `{:dependencies [[mvxcvi/puget "0.7.1"]
-                   [mvxcvi/whidbey "0.5.1"]]
+(defn- find-plugin-version
+  "Looks up the plugins in the project map and tries to find the version
+  specified for the given symbol. Returns nil if none matches."
+  [project plugin]
+  (try
+    (some (fn [[p v]] (when (= p plugin) v))
+          (:plugins project))
+    (catch Exception e
+      nil)))
 
-    :injections [(do (require 'whidbey.render)
-                     (alter-var-root
-                       #'whidbey.render/puget-options
-                       merge
-                       ~options))]
 
-    :repl-options {:nrepl-middleware [clojure.tools.nrepl.middleware.render-values/render-values]
-                   :nrepl-context {:interactive-eval {:renderer whidbey.render/render-str}}}})
+(defn- plugin-dependency
+  "Looks up the version for the given plugin, or sets it to `\"RELEASE\"`.
+  Returns a dependency vector entry."
+  [project plugin]
+  (vector
+   plugin
+   (or (find-plugin-version project plugin)
+       "RELEASE")))
+
+
+(defn- add-dependencies
+  "Adds some dependencies to the end of the current vector."
+  [project & deps]
+  (update-in project [:dependencies] concat deps))
+
+
+(defn- add-repl-init
+  "Adds the given repl initialization forms to the `:init` key in
+  `:repl-options`. Any existing initialization will happen before the
+  form is executed."
+  [project & forms]
+  (update-in project
+             [:repl-options :init]
+             (fn [current]
+               (if current
+                 `(do ~current ~@forms)
+                 `(do ~@forms)))))
+
+
+(defn- add-nrepl-middleware
+  "Adds the middleware identified by the given symbol to the *front* of the
+  current nrepl-middleware list in the project map."
+  [project sym]
+  (update-in project
+             [:repl-options :nrepl-middleware]
+             #(into [sym] %)))
+
+
+(defn- set-interactive-eval-renderer
+  "Sets the nrepl renderer for the interactive-eval context to the function
+  named by the given symbol."
+  [project sym]
+  (assoc-in project
+            [:repl-options :nrepl-context :interactive-eval :renderer]
+            sym))
 
 
 (defn middleware
+  "Rewrites the project to include Whidbey's customizations when the project
+  includes one of the target profiles."
   [project]
-  (let [options (:puget-options project)
-        included (:included-profiles (meta project))]
-    (if (some #{::profile} included)
-      project
-      (-> project
-          (project/add-profiles {::profile (whidbey-profile options)})
-          (project/merge-profiles [::profile])))))
+  ; TODO: stop accepting :puget-options eventually.
```

Comment:
Hmm, tempting - I might release this as 1.0.0 after a few more changes. Notably, I want to move the tagged-literal extensions for byte arrays and URIs from Puget to Whidbey and make them controllable with an option.